--- a/drivers/mtd/spi-nor/core.c
+++ b/drivers/mtd/spi-nor/core.c
@@ -2953,8 +2953,7 @@ void spi_nor_restore(struct spi_nor *nor)
 {
 	/* restore the addressing mode */
-	if (nor->addr_width == 4 && !(nor->flags & SNOR_F_4B_OPCODES) &&
-	    nor->flags & SNOR_F_BROKEN_RESET)
+	if (nor->addr_width == 4 && !(nor->flags & SNOR_F_4B_OPCODES))
 		nor->params->set_4byte_addr_mode(nor, false);
 
 	if (nor->flags & SNOR_F_SOFT_RESET)
